crates/context_aware_config/src/api/custom_types/mod.rs

```diff
@@ -0,0 +1,90 @@
+mod types;
+use std::str::FromStr;
+
+use crate::db::models::JsonSchemaTypes;
+use crate::db::schema::jsonschema_types::{self, dsl};
+use actix_web::web::{Json, Path};
+use actix_web::{delete, get, put, HttpResponse, Scope};
+use chrono::Utc;
+
+use diesel::query_dsl::methods::FilterDsl;
+use diesel::{ExpressionMethods, RunQueryDsl};
+use jsonschema::JSONSchema;
+use serde_json::json;
+use service_utils::service::types::DbConnection;
+use service_utils::{bad_argument, db_error, result as superposition};
+use superposition_types::User;
+
+use crate::api::custom_types::types::CreateTypeRequest;
+
+pub fn endpoints() -> Scope {
+    Scope::new("")
+        .service(list_types)
+        .service(create_or_update_type)
```

Comment:
Lets follow having create and update as separate APIs.

Reply:
@mahatoankitkumar In this case, there is a lot of code overlap. Should we still proceed with this? I found it redundant

Reply:
Yeah in future code becomes more complex as we keep adding features or changes to this. We noticed this with default_config create/update. And we followed having separate APIs from functions onwards to prevent this.
And for common code you can make functions for them.

Reply:
Agree - when we did `functions` we decided to have separate CREATE and UPDATE endpoints for clear behaviour separation.

Reply:
Sure, will add this.